internal/services/appservice/linux_function_app_slot_resource.go

```diff
@@ -344,7 +344,7 @@ func (r LinuxFunctionAppSlotResource) Create() sdk.ResourceFunc {
 			}
 
 			availabilityRequest := web.ResourceNameAvailabilityRequest{
-				Name: utils.String(functionAppSlot.Name),
+				Name: utils.String(fmt.Sprintf("%s-%s", functionAppId.SiteName, functionAppSlot.Name)),
```

Comment:
We should use the values from the ID here:
```suggestion
				Name: utils.String(fmt.Sprintf("%s-%s", id.SiteName, id.SlotName)),
```